src/main/java/net/dv8tion/jda/api/entities/channel/unions/DefaultGuildChannelUnion.java

```diff
@@ -0,0 +1,101 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.entities.channel.unions;
+
+import net.dv8tion.jda.api.entities.*;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A specialized union representing all channel types that can be used for the "default" channel for
+ * {@link Guild#getDefaultChannel()} or {@link Member#getDefaultChannel()}.
+ * <br>This class extends {@link StandardGuildChannel} and primarily acts as a discovery tool for
+ * developers to understand which channels might be returned as default channels.
+ *
+ * <br><br><b>Not all {@link StandardGuildChannel} channels can be used as a default channel!</b>
+ *
+ * <br><br>This interface represents the follow concrete channel types:
+ * <ul>
+ *     <li>{@link TextChannel}</li>
+ *     <li>{@link NewsChannel}</li>
+ * </ul>
+ */
+public interface DefaultGuildChannelUnion extends StandardGuildChannel
+{
+    /**
+     * Casts this union to a {@link TextChannel}.
+     * This method exists for developer discoverability.
+     *
+     * Note: This is effectively equivalent to using the cast operator:
+     * <pre><code>
+     * //These are the same!
+     * TextChannel channel = union.asTextChannel();
+     * TextChannel channel2 = (TextChannel) union;
+     * </code></pre>
+     *
+     * @throws IllegalStateException
+     *         If the channel represented by this union is not actually a {@link TextChannel}.
+     *
+     * @return The channel as a {@link TextChannel}
+     */
+    @Nonnull
+    TextChannel asTextChannel();
+
+    /**
+     * Casts this union to a {@link NewsChannel}.
+     * This method exists for developer discoverability.
+     *
+     * Note: This is effectively equivalent to using the cast operator:
+     * <pre><code>
+     * //These are the same!
+     * NewsChannel channel = union.asNewsChannel();
+     * NewsChannel channel2 = (NewsChannel) union;
+     * </code></pre>
+     *
+     * @throws IllegalStateException
+     *         If the channel represented by this union is not actually a {@link NewsChannel}.
+     *
+     * @return The channel as a {@link NewsChannel}
+     */
+    @Nonnull
+    NewsChannel asNewsChannel();
+
+    //TODO: add asForumChannel() (I think ForumChannels can be the default?)
+
+    /**
+     * Casts this union to a {@link StandardGuildMessageChannel}.
+     * <br>This only works for the following channel types represented by this union:
+     * <ul>
+     *     <li>{@link TextChannel}</li>
+     *     <li>{@link NewsChannel}</li>
+     * </ul>
```

Comment:
We shouldn't keep lists like this in getters, at most in the docs of the interface. But we don't really need these kinds of lists anyway since javadoc already provides it as "All Known Subinterfaces".

![image](https://user-images.githubusercontent.com/18090140/170248159-7c098649-7767-49d2-807a-f9b3badde465.png)

Reply:
The reasons these exist is because once ForumChannels and ViT exist, these lists wont be 1:1 with the implementers. Some channels represented by the union will be able to use the method and others wont.

Reply:
I still think we should not document the list on every getter, since that is a lot of things to remember to update.

Reply:
As a note, the All Known Subinterfaces wont work for this because the channel interfaces are not children to the union interfaces. i,e, this doesn't exist:
`TextChannel -> StandardChanneUnion -> Channel`

Unions are implemented only by the `impl` classes to avoid things like `TextChannel#asVoiceChannel()` being present.

Reply:
My assumption here is that when we add a new channel type to a union we're going to have to add the getter. At that point we'd update the list in the interface methods.

Reply:
I wouldn't even know which methods to look for when updating these lists because there are so many. I'd definitely say documenting on type and linking there is better for maintainability.

```
See {@link MessageChannelUnion} for a list of supported types. You can use {@link #getType} to check the type at runtime.
```

Reply:
Removed the lists